sonar-core/src/main/resources/org/sonar/core/component/db/ComponentMapper.xml

```diff
@@ -156,7 +156,7 @@
     </where>
   </select>
 
-  <select id="selectAllViewsAndSubViews" resultType="hashmap">
+  <select id="selectAllViewsAndSubViews" resultType="UuidWithProjectUuid">
```

Comment:
In fact no need to add an index on projects.qualifier : as the goal is to remove all views from the index that not exists in db, it would be better to search views from the list of view uuid found in the index.